src/main/kotlin/com/sourcegraph/cody/history/HistoryTree.kt

```diff
@@ -0,0 +1,84 @@
+package com.sourcegraph.cody.history
+
+import com.intellij.ui.treeStructure.Tree
+import com.sourcegraph.cody.history.listener.DeleteListener
+import com.sourcegraph.cody.history.listener.DoubleLeftClickListener
+import com.sourcegraph.cody.history.listener.EnterListener
+import com.sourcegraph.cody.history.listener.RightClickListener
+import com.sourcegraph.cody.history.state.ChatState
+import com.sourcegraph.cody.history.util.ToolbarDurationTextFormatter
+import java.awt.event.MouseEvent
+import java.time.LocalDateTime
+import java.time.temporal.ChronoUnit
+import javax.swing.tree.DefaultMutableTreeNode
+import javax.swing.tree.DefaultTreeModel
+
+class HistoryTree(onSelect: (ChatState) -> Unit, onDelete: (ChatState) -> Unit) : Tree() {
+
+  private val popup = HistoryPopupMenu(onSelect, onDelete, getSelected = { getLeafOrNull()!!.chat })
+
+  init {
+    isRootVisible = false
+    addMouseListener(DoubleLeftClickListener { invokeOnlyOnLeaf { onSelect(it) } })
+    addMouseListener(
+        RightClickListener { mouse -> invokeOnlyOnLeaf { popup.show(this, mouse.x, mouse.y) } })
+    addKeyListener(EnterListener { invokeOnlyOnLeaf { onSelect(it) } })
+    addKeyListener(DeleteListener { invokeOnlyOnLeaf { onDelete(it) } })
+  }
+
+  override fun getToolTipText(event: MouseEvent): String {
+    val leaf = getPathForLocation(event.x, event.y)?.lastPathComponent as? ChatLeafNode
+    if (leaf != null) {
+      val duration =
+          ToolbarDurationTextFormatter.formatDuration(
+              leaf.chat.latestMessage(), LocalDateTime.now())
+      return "Last updated: $duration ago"
+    }
+    return ""
```

Comment:
I think if there is no tooltip normally IJ APIs returns null.
Even this specific metod docs says:
`Returns:
a string containing the tooltip or null if event is null`